draft-ietf-tls-esni.md

```diff
@@ -396,25 +396,27 @@ reference an extension in ClientHelloOuter.
 
 When sending ClientHello, the client first computes ClientHelloInner, including
 any PSK binders. It then computes a new value, the EncodedClientHelloInner, by
-first making a copy of ClientHelloInner. It then MAY substitute extensions
-which it knows will be duplicated in ClientHelloOuter. To do so, the client
-removes and replaces extensions from EncodedClientHelloInner with a single
-"outer_extensions" extension. Removed extensions MUST be ordered consecutively
-in ClientHelloInner. The list of outer extensions, OuterExtensions, includes
-those which were removed from EncodedClientHelloInner, in the order in which
-they were removed.
+first making a copy of ClientHelloInner. It then clears the legacy_session_id
+field.
```

Comment:
Does this mean to send a 0-length `legacy_session_id`, or does this mean to send a `legacy_session_id` with the same length, but with all zeros? Presumably you mean the former?

Reply:
Reworded to be clearer.